open_spiel/games/chess/chess_board.h

```diff
@@ -131,10 +131,11 @@ inline constexpr std::array<Offset, 8> kKnightOffsets = {
 absl::optional<Square> SquareFromString(const std::string& s);
 
 // Forward declare ChessBoard here because it's needed in Move::ToSAN.
-template <uint32_t kBoardSize>
+template <uint32_t kBoardSize, bool kAllowCheck>
```

Comment:
The name is too generic. I'm not sure what would be better though. Maybe something like `kAllowKingToBeInCheck` ?

Also add an explanatory comment (along the lines of from wikipedia):

> A player is not told if their king is in check. Failing to move out of check, or moving into check, are both legal, and can obviously result in a capture and loss of the game.